docs/dashboard/dimensions-contexts-families.mdx

```diff
@@ -0,0 +1,93 @@
+type: explanation
+custom_edit_url: https://github.com/netdata/netdata/edit/master/docs/dashboard/dimensions-contexts-families.mdx
+---
+
+# Chart dimensions, contexts, and families
+
+While Netdata's charts require no configuration and are [easy to interact with](/docs/dashboards/interact-charts.mdx),
+they have a lot of underlying complexity. To meaningfully organize charts out of the box based on what's happening in
+your nodes, Netdata uses the concepts of **dimensions**, **contexts**, and **families**. 
+
+Understanding how these work will help you more easily navigate the dashboard, [write new
+alarms](/docs/monitor/configure-alars.md), or play around with the [API](/web/api/README.md).
+
+For a refresher on the anatomy of a chart, see [dashboards and charts](/docs/dashboards/dashboards-charts.mdx).
+
+## Dimension
+
+A **dimension** is a value that gets shown on a chart. The value can be raw data or calculated values, such as the
+average (the default), minimum, or maximum. These values can then be given any type of unit. For example, CPU
+utilization is represented as a percentage, disk I/O as `MiB/s`, and available RAM as an absolute value in `MiB` or
+`GiB`.
+
+Beneath every chart (or on the right-side if you [configure the
+dashboard](/docs/dashboards/customize.mdx#change-chart-legend-position)) is a legend of dimensions. When there are
+multiple dimensions, you'll see a different entry in the legend for each dimension.
+
+The **Apps CPU Time** chart (with the [context](#contexts) `apps.cpu`), which visualizes CPU utilization of
+different types of processes/services/applications on your node, always provides a vibrant example of a chart with
+multiple dimensions.
+
+![An example apps.cpu chart with many
+dimensions](https://user-images.githubusercontent.com/1153921/114207816-a5cb7400-9911-11eb-8800-06f60b745f9c.png)
+
+The chart shows 13 unique dimensions, such as `httpd` for the CPU utilization for web servers, `kernel` for anything
+related to the Linux kernel, and so on. In your dashboard, these specific dimensions will almost certainly be different.
+
+Dimensions can be [hidden](/docs/dashboards/interact-charts.mdx#show-and-hide-dimensions) to help you focus your
+attention.
+
+## Context
+
+A **context** is a way of grouping charts by the types of metrics collected and dimensions displayed. It's kind of like
+a machine-readable naming and organization scheme.
+
+For example, the **Apps CPU Time** has the context `apps.cpu`. A little further down on the dashboard is a similar
+chart, **Apps Real Memory (w/o shared)** with the context `apps.mem`. The `apps` portion of the context is the **type**,
+whereas anything after the `.` is specified either by the chart's developer or by the [family](#family). 
+
+Because these two charts share the `apps` type as part of their context, they're grouped together into the
+**Applications** section on the dashboard.
```

Comment:
I am not sure context affects grouping on the dashboard. 

Default is:
 - **type** is top level menu section.
 - **family** is subsection.

See type and family description in the [CHART](https://learn.netdata.cloud/docs/agent/collectors/plugins.d#chart).

Also, there are a bunch of tricks in the [main.js](https://github.com/netdata/netdata/blob/924bf118e59ddf72052b75f43c3eb7ef13afe693/web/gui/main.js#L1361-L1384) to group things _properly_ - otherwise, disks wouldn't be under the same top-level section 😄  

We mainly use context in alarms templates and for rendering `info` in the [dashboard_info.js](https://github.com/netdata/netdata/blob/master/web/gui/dashboard_info.js).

Reply:
I tried to explain this a bit more, but it's... odd.

When I'm done with all this user-facing stuff, someone (wink wink) should absolutely start creating some developer documentation around these gotchas and quirks.

Reply:
Well, check this comment, it is not gotchas 😄 
https://github.com/netdata/netdata/blob/924bf118e59ddf72052b75f43c3eb7ef13afe693/web/gui/main.js#L1361-L1364